src/services/base/base.service.js

```diff
@@ -530,14 +569,89 @@ let get_all_args = (req, args) => {
     return combined_args;
 };
 
-let check_fhir_mismatch = (cleaned, patched) => {
+const check_fhir_mismatch = (cleaned, patched) => {
     if (deepEqual(cleaned, patched) === false) {
         let diff = compare(cleaned, patched);
         logger.warn('Possible FHIR mismatch - ' + cleaned.resourceType + cleaned.id + ':' + cleaned.resourceType);
         logger.warn(diff);
     }
 };
 
+/**
+ * Checks whether the resource has any access codes that are in the passed in accessCodes list
+ * @param {string[]} accessCodes
+ * @param {string} user
+ * @param {string} scope
+ * @param {Resource} resource
+ * @return {boolean}
+ */
+const doesResourceHaveAnyAccessCodeFromThisList = (accessCodes, user, scope, resource) => {
+    // fail if there are no access codes
+    if (accessCodes.length === 0) {
+        let errorMessage = 'user ' + user + ' with scopes [' + scope + '] has no access scopes';
+        throw new ForbiddenError(errorMessage);
+    }
+    // see if we have the * access code
+    else if (accessCodes.includes('*')) {
+        // no security check since user has full access to everything
+        return true;
+    } else if (!resource.meta || !resource.meta.security) {
```

Comment:
More of a general question, are we requiring all resources to have security tags? If yes, what's the plan to retroactively fix the already generated ones

Reply:
yes.  the plan is to reload all the data except Questionnaire stuff since we're not working on vaccination.